File: Python/TwoPointer.py
def isPairExist(arr,target):
    arr.sort()
    myDict = {}
    i,j = 0, len(arr)-1
    for i in range(0,len(arr)-1):
        myDict[arr[i]] = i
    while i < j:
        sum = arr[i]+arr[j]
        if sum == target:
            index1 = myDict[arr[i]]
            index2 = myDict[arr[j]]
            return [index1,index2]
        if sum > target: j-=1
        if sum < target: i+=1
    return []


def main():
    arr = [-3,8,-9,3,18,7,-2,1,1,5,-6]
    target = 25
    ans = isPairExist(arr,target)
    print(ans)
main()

File: stanley/ece-eee/06-12-2024/TwoPointer.java
import java.util.Arrays;

class TwoPointer{
    public static void main(String[] args) {
        int[] arr = {-3,8,-2,9,7,6,12,-18,16,4};
        int target = 21;
        boolean ans = isPairExist(arr,target);
        System.out.println(ans);
    }
    private static boolean isPairExist(int[] arr, int target){
        Arrays.sort(arr);
        int i = 0, j = arr.length-1;
        while (i < j) {
            int sum = arr[i]+arr[j];
            if(sum == target) return true;
            if(sum > target) j--;
            if(sum < target)i++;
        }
        return false;
    }
}